include/git2/sys/odb_backend.h

```diff
@@ -100,6 +100,19 @@ struct git_odb_backend {
 	 * itself). An odb backend implementation must provide this function.
 	 */
 	void GIT_CALLBACK(free)(git_odb_backend *);
+
+	/**
+	 * The subdirectory of the .git repository to use as storage.
+	 *
+	 * "object" is the git standard, but can be NULL to signal that the backend
+	 * does not have a filesystem location (a.k.a is in-memory).
+	 *
+	 * Note that at the moment, adding a non-default backend to the odb will
+	 * result in a repository where GIT_REPOSITORY_ITEM_OBJECTS cannot make
+	 * sense, and will thus be disabled. Passing "objects" here is thus
+	 * reserved to backends that can make sense of its contents.
+	 */
+	const char *location;
```

Comment:
I was attempting *something* with that, but I'm not quite sure now. I felt that knowing that we're using our own "standard" backends was something generally useful. And I felt that knowing if a backend has or has no filesystem location was as well, and that this aspect could be used as part of the definition of what is an in-memory repository. You could well imagine a backend that uses the filesystem, but it's not ".git/objects"-compatible: how can you signal that to the repository, so that it knows to report that its `GIT_REPOSITORY_ITEM_OBJECTS` contents is non-standard ? Not sure where I was going with this though…

Reply:
I agree that this would be useful. But that'd require a new callback in the backends, as we generally shouldn't assume anything of backends. Might very well be (well, probably not) that somebody has implemented his own implementation of a ".git/objects" backend that is uber fast due to it using platform-specific things.